lib/request.js

```diff
@@ -0,0 +1,34 @@
+module.exports = request
+
+var Promise = require('./utils/promise')
+
+var internals = module.exports.internals = {}
+internals.request = require('./utils/request')
+
+function request (state, options) {
+  if (!options) {
+    return Promise.reject(new Error('hoodie.request: URL or options argument must be defined'))
+  }
+
+  var requestOptions = {
+    url: options.url || options
+  }
+
+  if ((/^\/(?!\/)/).test(requestOptions.url)) {
```

Comment:
I would suggest using the 'url' module here. `if (!url.parse(requestOptions.url).protocol)`